src/watertap_contrib/reflo/unit_models/crystallizer_effect.py

```diff
@@ -0,0 +1,430 @@
+        self.solids.temperature.setub(1000)
+        self.vapor.temperature.setub(1000)
+        self.pure_water.temperature.setub(1000)
+
+        self.steam_pressure = Param(
+            initialize=3,
+            mutable=True,
+            units=pyunits.bar,
+            doc="Steam pressure (gauge) for crystallizer heating: 3 bar default based on Dutta example",
+        )
+
+        self.efficiency_pump = Param(
+            initialize=0.7,
+            mutable=True,
+            units=pyunits.dimensionless,
+            doc="Crystallizer pump efficiency - assumed",
+        )
+
+        self.pump_head_height = Param(
+            initialize=1,
+            mutable=True,
+            units=pyunits.m,
+            doc="Crystallizer pump head height -  assumed, unvalidated",
+        )
+
+        self.energy_flow_superheated_vapor = Var(
+            initialize=1e5,
+            bounds=(-5e6, 5e6),
+            units=pyunits.kilowatt,
+            doc="Energy that could be supplied from vapor",
+        )
+
+        self.delta_temperature_in = Var(
+            self.flowsheet().time,
+            initialize=35,
+            bounds=(None, None),
+            units=pyunits.degK,
+            doc="Temperature difference at the inlet side",
+        )
+
+        self.delta_temperature_out = Var(
+            self.flowsheet().time,
+            initialize=35,
+            bounds=(None, None),
+            units=pyunits.degK,
+            doc="Temperature difference at the outlet side",
+        )
+
+        delta_temperature_chen_callback(self)
+
+        self.heat_exchanger_area = Var(
+            initialize=1000.0,
+            bounds=(0, None),
+            units=pyunits.m**2,
+            doc="Heat exchanger area",
+        )
+
+        self.overall_heat_transfer_coefficient = Var(
+            initialize=0.1,
+            bounds=(0, None),
+            units=pyunits.kilowatt / pyunits.m**2 / pyunits.degK,
+            doc="Overall heat transfer coefficient for heat exchangers",
+        )
+
+        @self.Constraint()
+        def eq_pure_vapor_flow_rate(b):
+            return (
+                b.properties_pure_water[0].flow_mass_phase_comp["Vap", "H2O"]
+                == b.properties_in[0].flow_mass_phase_comp["Vap", "H2O"]
+            )
+
+        @self.Constraint(doc="Thermal energy in the vapor")
+        def eq_vapor_energy_constraint(b):
+            return b.energy_flow_superheated_vapor == (
+                b.properties_vapor[0].flow_mass_phase_comp["Vap", "H2O"]
+                * (
+                    b.properties_pure_water[
+                        0
+                    ].dh_vap_mass_solvent  # Latent heat from the vapor
+                    + b.properties_vapor[0].enth_mass_solvent["Vap"]
+                    - b.properties_pure_water[0].enth_mass_solvent["Vap"]
+                )
+            )
+
+        self.del_component(self.eq_p_con1)
+        self.del_component(self.eq_p_con2)
```

Comment:
The main thing that changed were the units on enthalpy I think, from kJ/kg to J/kg, so increasing enthalpy bound values and initial values by 3 orders of magnitude. Scaling was adjusted in accordance as well. However, some nonobvious changes had to also be implemented to the scaling routine in order to get the model to converge and tests to pass again: https://github.com/watertap-org/watertap/pull/1462/files#diff-e2613f205b01d4e211ff0cb63a36c0e12136e96c1e13eef0dc47bf078b0b9a0aR766-R777

I am guessing you and @Zhuoran29 probably identified these changes already and still had issues with implementation, but just in case you didn't, hopefully some of that will be useful for future probing.

Reply:
Yes exactly. We identified the changes and it just wasn't cooperating. It is something we are going to have to reconcile at some point when we point to a newer watertap release but not in this PR.